modules/core/src/shaderlib/project/viewport-uniforms.js

```diff
@@ -153,6 +153,58 @@ function calculateMatrixAndOffset({
   };
 }
 
+export function getCameraPositionFromViewport({
```

Comment:
Is this just duplicating code in this file? At least reuse this for project module's uniform generation?

Reply:
You should be able to import the `project` module and call `project.getUniforms` without this new utility function.

Reply:
I made this new API according to ib's review comments, I can revert this change if it is not good. Orginially I use the existing API 'getUniformsFromViewport', which way do you prefer?

Reply:
Going to revert back to 'getUniformsFromViewport' solution.